docs/release_note_generator/main.py

```diff
@@ -0,0 +1,71 @@
+# Usage:
+# GITHUB_TOKEN=<token> python main.py --old <old> --new <new> --path <path>
+#
+# The above command will create release notes for changes between the <old> commit and the <new> commit.
+# Only commits that change files in <path> will be included. A github token is required to avoid rate limits.
+
+import argparse
+import os
+import requests
+
+GITHUB_TOKEN_ENV_VAR = "GITHUB_TOKEN"
+COMMITS_ENDPOINT = "https://api.github.com/repos/dydxprotocol/v4-chain/commits"
+GET_COMMIT_ENDPOINT = "https://api.github.com/repos/dydxprotocol/v4-chain/commits/%s"
+LIST_COMMIT_PULLS_ENDPOINT = "https://api.github.com/repos/dydxprotocol/v4-chain/commits/%s/pulls"
+
+
+def get_commit(session, commit_sha):
+    r = session.get(GET_COMMIT_ENDPOINT % commit_sha)
+    return r.json()
+
+# Return a string that is a markdown list entry. It should summarize the changes in this commit.
+def commit_to_entry(session, commit_json):
+    r = session.get(LIST_COMMIT_PULLS_ENDPOINT % commit_json["sha"])
+    pulls = r.json()
+
+    if len(pulls) > 1:
+        # I don't know when a commit would have more than one PR. If this happens, report a bug.
+        raise NotImplementedError("Commit unexpectedly had more than one associated PR. Please file a bug.")
+    elif len(pulls) == 1:
+        # If there is an associated PR, use PR title and link to PR.
+        pull = pulls[0]
+        return "- %s ([#%s](%s))" % (pull["title"], pull["number"], pull["html_url"])
```

Comment:
is it useful to also pull the labels?

Reply:
In the future I was thinking to use labels to organize these into sections. For now I don't think it would be very useful, so it'll have to be done manually.

Reply:
Can we create a ticket and add a TODO here to do that? What you outlined seems useful!